src/codegate/workspaces/crud.py

```
@@ -77,3 +78,20 @@ async def activate_workspace(self, workspace_name: str) -> bool:
         db_recorder = DbRecorder()
         await db_recorder.update_session(session)
         return True
+
+    async def update_workspace_system_prompt(
+        self, workspace_name: str, sys_prompt_lst: List[str]
+    ) -> Optional[Workspace]:
+        selected_workspace = await self._db_reader.get_workspace_by_name(workspace_name)
```

can we throw an exception instead of returning `None`?

---

Yes, nice suggestion. Done so